src/Exceptions/PermissionException.php

```diff
@@ -0,0 +1,10 @@
+<?php
+
+namespace CodeIgniter\Shield\Exceptions;
+
+use CodeIgniter\Exceptions\ExceptionInterface;
+use RuntimeException;
+
+class PermissionException extends RuntimeException implements ExceptionInterface
```

Comment:
Instead of new exception classes, you can add methods to `CodeIgniter\Shield\Authorization\AuthorizationException` to cover your needs.

Reply:
That's it?

```php
<?php

namespace CodeIgniter\Shield\Exceptions;

use CodeIgniter\Shield\Authorization\AuthorizationException;

class PermissionException extends AuthorizationException
{
}

```

Reply:
or it is not necessary `GroupException `and `PermissionException`, only `AuthorizationException`

Reply:
I believe that is exactly what Lonnie meant.